src/Symfony/Component/Cache/Adapter/CouchbaseBucketAdapter.php

```diff
@@ -0,0 +1,226 @@
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Cache\Adapter;
+
+use Couchbase\Bucket;
+use Symfony\Component\Cache\Exception\CacheException;
+use Symfony\Component\Cache\Exception\InvalidArgumentException;
+use Symfony\Component\Cache\Marshaller\DefaultMarshaller;
+use Symfony\Component\Cache\Marshaller\MarshallerInterface;
+
+/**
+ * Class CouchbaseBucketAdapter.
+ */
+class CouchbaseBucketAdapter extends AbstractAdapter
+{
+    private const THIRTY_DAYS_IN_SECONDS = 2592000;
+    private const MAX_KEY_LENGTH = 250;
+    private const KEY_NOT_FOUND = 13;
+
+    /** @var Bucket */
+    private $bucket;
+
+    /** @var MarshallerInterface */
+    private $marshaller;
+
+    public function __construct(\CouchbaseCluster $client, string $bucket, string $namespace = '', int $defaultLifetime = 0, array $options = [])
+    {
+        $this->init($client, $bucket, $namespace, $defaultLifetime, $options);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function createConnection($servers, array $options = []): \CouchbaseCluster
+    {
+        if (\is_string($servers)) {
+            $servers = [$servers];
+        } elseif (!\is_array($servers)) {
+            throw new InvalidArgumentException(sprintf('CouchbaseAdapter::createClient() expects array or string as first argument, %s given.', \gettype($servers)));
+        }
+
+        if (!static::isSupported()) {
+            throw new CacheException('Couchbase >= 2.6.0 is required');
+        }
+
+        set_error_handler(function ($type, $msg, $file, $line) { throw new \ErrorException($msg, 0, $type, $file, $line); });
+
+        $newServers = [];
+        try {
+            $username = $options['username'] ?? '';
+            $password = $options['password'] ?? '';
+
+            foreach ($servers as $dsn) {
+                if (\is_array($dsn)) {
+                    continue;
+                }
+
+                if (0 !== strpos($dsn, 'couchbase:')) {
+                    throw new InvalidArgumentException(sprintf('Invalid Couchbase DSN: %s does not start with "couchbase:"', $dsn));
+                }
+
+                preg_match('/^(?<protocol>couchbase(s)?):\/\/((?<username>[^@:]*+):(?<password>[^@]*+)@)?(?<cluster>.+)(\/.*)?/m', $dsn, $matchers);
+
+                $username = $matchers['username'] ?: $username;
+                $password = $matchers['password'] ?: $password;
+                $newServers += explode(',', $matchers['cluster']);
+            }
+
+            $servers = $newServers ?? $servers;
+            $connectionString = $matchers['protocol'].'://'.implode(',', $servers);
+
+            $client = new \CouchbaseCluster($connectionString);
+            $client->authenticateAs($username, $password);
+
+            return $client;
+        } finally {
+            restore_error_handler();
+        }
+    }
+
+    /**
+     * @return bool
```

Comment:
We don't add docblock when they don't add to the signature of methods, please remove them all I think.

Reply:
Ok.